interview-questions.md

```diff
@@ -8,62 +8,61 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 1. What is iteration?
 
-  Your answer:
+  Your answer: Iteration is any method or process that allows a function or code to be executed throughout the entire length of a string, array, or object. For instance, for loop or .map() are ways to iterate across a parameter. Be aware of the datatype because each respond to a certain style of iteration.
 
-  Researched answer:
+  Researched answer: Iteration is the process of executing code until a condition is met.
 
 
 
 2. The JavaScript built in method .map() takes predefined parameters. What are they? Which are required and which are optional?
 
-  Your answer:
+  Your answer: .map takes value, index, an array as predefined parameters. All are optional to use; however, one needs to be designated. (Just had an epiphany about errors I have received using .map() before.)
 
-  Researched answer:
+  Researched answer: .map() is not concerned with the name of parameters; however, the order of the parameters will always be the same---(value, index, array). The value is a required parameter while index and array are optional.
 
 
 
 3. What is object destructuring?
 
-  Your answer:
-
-  Researched answer:
+  Your answer: Object destructuring is a method that allows keys and values in an object to be called upon by their value or index.
 
+  Researched answer: Object destructuring is a special technique to assign variables. In other words, breaking the properties of an object into separate, recallable variables. This technique comes in handy when coding objects with nested properties.
 
 
 4. What is the difference between a function and a method?
 
-  Your answer:
+  Your answer: The difference between a function and a method!!! Well, a function is the executable code that uses a series of methods and variables to perform an action and produce an output based on information introduced externally. A method is the structure or key words that help do the heavy lifting in the code. Methods allow the code to be easier to read and less wordy. This feature is wonderful when a person is troubleshooting errors or trying to update a code block.
 
-  Researched answer:
+  Researched answer: Functions are the hierarchy of shortcuts. They can be rendered by name and can standalone. The data passed through the function is limited by the approved parameters that will be manipulated the code block executed through the function. Methods are tools that can be used by a function. However, methods oare executed once the name of the object it is assigned to is called. A method is associated the property of an object.
```

Comment:
And once step further: a method is a function that belongs to **and** acts upon an object.